Application/IMessageRepository.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Domain;
+
+namespace Application
+{
+    public interface IMessageRepository
+    {
+        void AddMessages(List<Message> messages);
+        List<Message> GetMessagesBetweenTwoUsers(int idSender, int idReceiver);
```

Comment:
Consider using IEnumerable instead of the list when you will start using SQL database.
https://docs.microsoft.com/en-us/dotnet/standard/collections/when-to-use-generic-collections

Reply:
I'm not exactly sure how to use IEnumerable instead of List.
Should I replace every instance of List<> with IEnumerable<> and work with IEnumerables or should I use IEnumerable to extract the data from the database, convert it to List and then work with List?
The latter would be easier to implement because then I won't have to refactor so much code but if there's performance advantages, then I'll replace everything with IEnumerable.